pallets/contracts/src/lib.rs

```diff
@@ -315,12 +325,63 @@ decl_module! {
             ensure!(Identity::<T>::has_valid_cdd(new_owner), Error::<T>::NewOwnerIsNotCDD);
 
             // Change the `owner` variable value to the given did.
-            <TemplateMetaDetails<T>>::mutate(&code_hash, |meta_details| meta_details.owner = new_owner);
+            <TemplateInfo<T>>::mutate(&code_hash, |template_details| template_details.owner = new_owner);
 
             // Emit event.
             Self::deposit_event(RawEvent::TemplateOwnershipTransferred(did, code_hash, new_owner));
             Ok(())
         }
+
+
+        /// Change the usage fee & the instantiation fee of the smart extension template
+        ///
+        /// # Arguments
+        /// * origin - Only owner of template is allowed to execute the dispatchable.
+        /// * code_hash - Unique hash of the smart extension template.
+        /// * new_instantiation_fee - New value of instantiation fee for the smart extension template.
+        /// * new_usage_fee - New value of usage fee for the smart extension template.
+        #[weight = 1000_000_000]
+        pub fn change_template_fees(origin, code_hash: CodeHash<T>, new_instantiation_fee: Option<BalanceOf<T>>, new_usage_fee: Option<BalanceOf<T>>) -> DispatchResult {
+            // Ensure whether the extrinsic is signed & validate the `code_hash`.
+            let (did, _) = Self::ensure_signed_and_template_exists(origin, code_hash)?;
+
+            // Update the fees
+            if let Some(usage_fee) = new_usage_fee {
+                // Access the current usage fee.
+                let old_usage_fee = Self::get_metadata_of(code_hash).usage_fee;
+                // Update the usage fee for a given code hash.
+                <MetadataOfTemplate<T>>::mutate(&code_hash, |metadata| metadata.usage_fee =  usage_fee);
```

Comment:
This can be optimized to 
```
let old_usage_fee = <MetadataOfTemplate<T>>::mutate(&code_hash, |metadata| core::mem::replace(metadata.usage_fee, usage_fee));
```